app/src/main/java/org/schabi/newpipe/settings/NewPipeSettings.java

```diff
@@ -50,6 +55,32 @@ public static void initSettings(final Context context) {
         getAudioDownloadFolder(context);
     }
 
+    /**
+     * Backward-compatibility: convert boolean search suggestions value to new multi-select list.
+     * @param context the context to use to get shared preferences
+     */
+    private static void migrateSearchSuggestionsIfNeeded(final Context context) {
```

Comment:
I think we should do the migration process a little different. First of all, it is good to store the last used/started NewPipe version (or better version code). That should also help us when creating a welcome tour and showing changelogs in future versions.
Depending on the last used version, we could do something like this:
``` Java
int lastStartedVersionCode = preferences.getInt(context.getString(R.string.last_started_newpipe_version_code), -1);
if (lastStartedVersionCode < 951) { // version code for 0.19.6
    // do migration here or call a method
}
// register later migrations here
```
The only thing we should figure out is how to differentiate between a new installation and a user updating. Both have not set `last_started_newpipe_version_code`, but a new installation does not need to run migrations. @wb9688 @B0pol @Stypox Any thoughts on this or suggestions for a different structure of the migration process?
 
We should update the last used version code if necessary at the end of https://github.com/TeamNewPipe/NewPipe/blob/049c8f70cdb12fa95ce6d61bfc70e64584fbf063/app/src/main/java/org/schabi/newpipe/App.java#L85

Reply:
Done